### PR TITLE
Refactor hub endpoints

### PR DESCRIPTION
**Pull request by audreylim**

### CODE REVIEW

File: data/hub.go [hunk 1]
```diff
@@ -1,82 +1,88 @@
...
+	if err == sql.ErrNoRows {
+		return &Error{"record_not_found", "hub not found"}
+	}
+	return err
+}
+
+func (h *Hub) Delete(db *sqlx.DB) error {
+	nstmt, err := db.PrepareNamed(`DELETE FROM hubs
+	WHERE slug = (:slug)
+	RETURNING *;
```
Comment (laktek) [inline data/hub.go:77]: returning not needed.

Comment (audreylim) [inline data/hub.go:77]: oh ok, what should be in the json response?

File: data/hub_test.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+	h := &data.Hub{
+		Slug:   "earthworm",
+		UserID: 1,
+	}
+	if err := h.Insert(db); err != nil {
+		t.Error("Failed to insert h to db: %v", h)
+	}
+
+	// query for the inserted hub by hub
```
Comment (laktek) [inline data/hub_test.go:144]: copy-pasta comment ;)

Comment (audreylim) [inline data/hub_test.go:144]: haha deleted

File: data/hub.go (hunk 1 above)
Comment (laktek) [inline data/hub.go:77]: hmm..`DELETE` also returns column values similar to `SELECT` then it should be alright.

Comment (audreylim) [inline data/hub.go:77]: hm so I just return deleted hub slug in json or keep it this way?

Comment (laktek) [inline data/hub.go:77]: keep it this way

**Pull request merged**